Client/Source/Handler.h

```diff
@@ -0,0 +1,260 @@
+#ifndef HANDLER_H
+#define HANDLER_H
+
+#include <string>
+#include <unordered_map>
+
+#include "Utilities.h"
+
+
+namespace vasily
+{
+
+/**
+ * \brief Structure used to contains additional input data from client.
+ */
+struct ParsedResult
+{
+    /**
+     * \brief Number used to change coordinates.
+     */
+    int                    coefficient        = 1;
+
+    /**
+     * \brief Number of iterations for circlic movement.
+     */
+    int                    numberOfIterations = 1;
+
+    /**
+     * \brief Additional flag to check correctness of coordinates.
+     */
+    bool                   isCorrect          = true;
+
+    /**
+     * \brief List of parsed points.
+     */
+    std::vector<RobotData> points             = {};
+};
+
+/**
+ * \brief Additional helper which provides parsing input data from client.
+ */
+class Handler
+{
+public:
+    /**
+     * \brief Array of moves states.
+     */
+    enum class State
+    {
+        DEFAULT,
+
+        COORDINATE_TYPE,
+        FULL_CONTROL,
+        CIRCLIC,
+        PARTIAL,
+        POSITIONAL,
+        HOME,
+        FROM_FILE,
+
+        FORWARD,
+        BACK,
+        
+        LEFT,
+        RIGHT,
+        
+        UP,
+        DOWN,
+
+        ROLL_PLUS,
+        ROLL_MINUS,
+
+        PITHCH_PLUS,
+        PITHCH_MINUS,
+
+        YAW_PLUS,
+        YAW_MINUS
+    };
+
+    /**
+     * \brief Input mode.
+     */
+    enum class Mode
+    {
+        READING,
+        COMMAND
+    };
+
+    /**
+     * \brief Coordinate name, just to avoid "magic numbers".
+     */
+    static constexpr std::size_t X = 0;
+
+    /**
+     * \brief Coordinate name, just to avoid "magic numbers".
+     */
+    static constexpr std::size_t Y = 1;
+    
+    /**
+     * \brief Coordinate name, just to avoid "magic numbers".
+     */
+    static constexpr std::size_t Z = 2;
+    
+    /**
+     * \brief Coordinate name, just to avoid "magic numbers".
+     */
+    static constexpr std::size_t W = 3;
+    
+    /**
+     * \brief Coordinate name, just to avoid "magic numbers".
+     */
+    static constexpr std::size_t P = 4;
+    
+    /**
+     * \brief Coordinate name, just to avoid "magic numbers".
+     */
+    static constexpr std::size_t R = 5;
```

Comment:
**(optional):** Why you don't use enum here?

Reply:
Timur and I already have a discussion at this point. It's because of the ugly rule "You can't use old-plain C-style enum". I suppose we need to remove this rule.

Reply:
And I cannot to write every time:
`_robotData.coordinates.at(static_cast<std::size_t>(Handler::Coordinate::Y)) += 500;`